cluster/src/dunit/scala/io/snappydata/externalstore/ExecutorMessageDUnitTest.scala

```diff
@@ -0,0 +1,165 @@
+package io.snappydata.externalstore
+
+import java.io.{File, IOException}
+import java.util.Properties
+
+import com.pivotal.gemfirexd.FabricService
+import io.snappydata.{Locator, ServiceManager}
+import io.snappydata.cluster.{ClusterManagerTestBase, ExecutorInitiator}
+import io.snappydata.test.dunit.{SerializableRunnable, DistributedTestBase}
+import io.snappydata.test.dunit.DistributedTestBase.WaitCriterion
+
+import org.apache.spark.Logging
+import org.apache.spark.sql.SnappyContext
+
+class ExecutorMessageDUnitTest(s: String) extends ClusterManagerTestBase(s) with Logging {
+
+  def testStore01BlockMapUpdatesWithExecutorDown(): Unit = {
+    import ExecutorMessageDUnitTest._
+    executeSomething()
+    verifyMap("stopExecutor")
+    restartExecutors(locatorPort, bootProps)
+    getLogWriter.info("testStore01BlockMapUpdatesWithExecutorDown() Successful")
+  }
+
+  // TODO The next test that runs after this one hangs/fails because we stop the
+  // server process in this test. Enable it after resolving the issue.
+  // When run individually, this test passes.
+  def _testStore02BlockMapUpdatesWithNodeDown(): Unit = {
```

Comment:
This was missing and hiding a small issue too. Thanks for pointing this out. Will update the code.